docs: add 4.0 release notes for SD and disk

danieldegrasse opened the pull request:
Add release notes for the disk and SD subsystems. Note that the SD section has been removed for this release, as no significant changes were made to the subsystem since 3.7.

danieldegrasse committed:
doc: release-notes-4.0: add release notes for disk

Add release notes for the disk driver subsystem

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
doc: release-notes-4.0: add SDHC release notes

Add release notes for SD host controllers for 4.0

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

dkalowsk commented on doc/releases/release-notes-4.0.rst:
```diff
@@ -1068,8 +1068,6 @@ Libraries / Subsystems
 
 * Random
 
-* SD
```
@decsny and I were just talking in another PR that it may be better to just leave the section in and say "No changes in this release" or something like that.  Thoughts?

danieldegrasse replied:
I was also considering this. I'm fine with that approach- I'll update this PR

danieldegrasse committed:
doc: release-notes-4.0: add release notes for SD subsystem

No significant changes have been made to the SD subsystem since 3.7. Add
a note documenting this for the 4.0 release.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>
